src/adm/filter.rs

```diff
@@ -107,6 +119,84 @@ impl AdmFilter {
         l_sentry::report(&merged_tags, sentry::event_from_error(error));
     }
 
+    /// check to see if the bucket has been modified since the last time we updated.
+    pub async fn requires_update(&self) -> HandlerResult<bool> {
+        // don't update non-bucket versions (for now)
+        if !self.is_cloud() {
+            return Ok(false);
+        }
+        if let Some(bucket) = &self.source_url {
+            let host = match bucket.host() {
+                Some(v) => v,
+                None => {
+                    return Err(HandlerError::internal(&format!(
+                        "Missing bucket Host {:?}",
+                        self.source
+                    )))
+                }
+            }
+            .to_string();
+
+            let obj: cloud_storage::Object = cloud_storage::Object::read(&host, bucket.path())
+                .await
+                .map_err(|e| {
+                    HandlerError::internal(&format!(
+                        "Could not read bucket {:?}, {:?}",
+                        self.source, e
+                    ))
+                })?;
+            if let Some(updated) = self.last_updated {
+                // if the bucket is older than when we last checked, do nothing.
+                return Ok(updated <= obj.updated);
+            };
+            return Ok(true);
+        }
+        Ok(false)
+    }
+
+    /// Try to update the ADM filter data from the remote bucket.
+    pub async fn update(&mut self) -> HandlerResult<()> {
+        if let Some(bucket) = &self.source_url {
+            let adm_settings = AdmSettings::from_settings_bucket(bucket)
+                .await
+                .map_err(|e| {
+                    HandlerError::internal(&format!(
+                        "Invalid bucket data in {:?}: {:?}",
+                        self.source, e
+                    ))
+                })?;
+            for (adv, setting) in adm_settings.advertisers {
```

Comment:
This should also handle an advertiser entry being removed upstream

Reply:
added `delete` flag to ADMFilter settings.